# DP: Add mudflap support for varargs

From: Michael Matz <matz@suse.de>
Subject: Fix pass49-frag (mudflap support for varargs)

	* builtins.c (build_va_arg_indirect_ref): Use
	build_simple_mem_ref_loc.

Index: builtins.c
===================================================================
--- a/src/gcc/builtins.c	(revision 171675)
+++ b/src/gcc/builtins.c	(working copy)
@@ -4748,7 +4748,7 @@ std_gimplify_va_arg_expr (tree valist, t
 tree
 build_va_arg_indirect_ref (tree addr)
 {
-  addr = build_fold_indirect_ref_loc (EXPR_LOCATION (addr), addr);
+  addr = build_simple_mem_ref_loc (EXPR_LOCATION (addr), addr);
 
   if (flag_mudflap) /* Don't instrument va_arg INDIRECT_REF.  */
     mf_mark (addr);
